web/src/main/java/se/inera/intyg/minaintyg/certificate/service/CertificateStatusServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package se.inera.intyg.minaintyg.certificate.service;
+
+import org.springframework.stereotype.Service;
+import se.inera.intyg.minaintyg.integration.api.certificate.dto.CertificateRecipient;
+import se.inera.intyg.minaintyg.integration.api.certificate.dto.CertificateRelation;
+import se.inera.intyg.minaintyg.integration.api.certificate.dto.CertificateRelationType;
+import se.inera.intyg.minaintyg.integration.api.certificate.dto.CertificateStatusType;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class CertificateStatusServiceImpl implements CertificateStatusService {
+
+    @Override
+    public List<CertificateStatusType> get(List<CertificateRelation> relations,
+                                           CertificateRecipient recipient,
+                                           LocalDateTime issued) {
+
+        List<Optional<CertificateStatusType>> statuses;
+
+        if (relations == null) {
+            statuses = new ArrayList<>();
+        } else {
+            statuses = relations
+                    .stream()
+                    .map(this::renewed)
+                    .collect(Collectors.toList());
+        }
+
+        statuses.add(sent(recipient));
+        statuses.add(newStatus(issued));
+
+        return statuses
+                .stream()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
```

Comment:
Can this be done in an easier way instead of creating the list and adding elements?